scripts/build/Jenkins/Jenkinsfile

```diff
@@ -378,6 +378,25 @@ def TestMetrics(Map pipelineConfig, String workspace, String branchName, String
     }
 }
 
+def BenchmarkMetrics(Map pipelineConfig, String workspace, String branchName, String outputDirectory) {
+    catchError(buildResult: null, stageResult: null) {
+        def cmakeBuildDir = [workspace, ENGINE_REPOSITORY_NAME, outputDirectory].join('/')
+        dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
+            checkout scm: [
+                $class: 'GitSCM',
+                branches: [[name: '*/main']],
+                extensions: [
+                    [$class: 'AuthorInChangelog'],
+                    [$class: 'RelativeTargetDirectory', relativeTargetDir: 'mars']
+                ],
+                userRemoteConfigs: [[url: "${env.MARS_REPO}", name: 'mars', credentialsId: "${env.GITHUB_USER}"]]
```

Comment:
In case this Jenkinsfile may run on a Jenkins environment that doesn't define env.MARS_REPO, I recommend to add a if check and only run these commands when env.MARS_REPO is defined.

Reply:
Yep, this is already done before this function is called:
if (env.MARS_REPO && params && params.containsKey('TEST_METRICS') && params.TEST_METRICS == 'True')